src/utils/middleware.ts

```diff
@@ -48,6 +48,26 @@ export interface ISpidStrategyOptions {
   sp: SamlConfig;
 }
 
+export function bindSpidStrategyOptions(
+  samlConfig: SamlConfig,
+  serviceProviderConfig: IServiceProviderConfig,
+  idpOptionsRecord: Record<string, IDPEntityDescriptor>
+): ISpidStrategyOptions {
+  return {
+    idp: idpOptionsRecord,
+    sp: {
+      ...samlConfig,
+      attributes: {
+        attributes: serviceProviderConfig.requiredAttributes,
+        name: "Required attributes"
+      },
+      identifierFormat: "urn:oasis:names:tc:SAML:2.0:nameid-format:transient",
+      organization: serviceProviderConfig.organization,
+      signatureAlgorithm: "sha256"
+    } as SamlConfig
```

Comment:
why this cast ? (add a comment)

Reply:
Fixed in a4bdb13
The `ISpidStrategyOptions` was extended to avoid casting